src/clr-angular/forms/datalist/datalist-container.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, ContentChild } from '@angular/core';
+import { ControlClassService } from '../common/providers/control-class.service';
+import { LayoutService } from '../common/providers/layout.service';
+import { DynamicWrapper } from '../../../clr-angular/utils/host-wrapping/dynamic-wrapper';
+import { DatalistControllerService } from './providers/datalist-controller.service';
+import { ControlIdService } from '../common/providers/control-id.service';
+import { FocusService } from '../common/providers/focus.service';
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { NgControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { ClrLabel } from '../common/label';
+
+@Component({
+  selector: 'clr-datalist-container',
+  template: `
+    <ng-content select="label"></ng-content>
+    <label *ngIf="!label && addGrid()"></label>
+    <div class="clr-control-container" [ngClass]="controlClass()">
+      <div class="clr-input-wrapper">
+        <div class="clr-input-group" [class.clr-focus]="focus">
+          <ng-content select="[clrDatalist]"></ng-content>
+          <datalist id="data-list">
```

Comment:
I believe we will need unique id here. Either generate one or take it from the user-defined input[list] attribute.

Reply:
added an auto generated way.